src/prefect/tasks/kafka/kafka.py

```diff
@@ -0,0 +1,211 @@
+    ) -> List[bytes]:
+        """
+        Run method for this Task. Invoked by calling this Task after initialization within a
+        Flow context, or by using `Task.bind`.
+        Args:
+        - bootstrap_servers (str:, required): comma separated host and port pairs that are the
+            addresses of kafka brokers
+        - group_id (str:, required): name of the consumer group the consumer will belong to
+        - topic (List[str], required): list of topic names to consume messages from
+        - request_timeout (float, optional): Maximum time to block waiting for message, event or callback
+        - auto_offset_reset (str, optional): configurable offset reset policy
+        - message_consume_limit (int, optional): max number of messages to consume before closing
+            the consumer
+        - **consumer_options (Any, optional): additional keyword arguments to pass to
+            confluent_kafka's `Consumer` init config
+        Returns:
+            - List of consumed messages
+        """
+        consumer = confluent_kafka.Consumer(
+            {
+                "bootstrap.servers": bootstrap_servers,
+                "group.id": group_id,
+                "auto.offset.reset": auto_offset_reset,
+                **consumer_options,
+            }
+        )
+        consumer.subscribe(topic)
+
+        messages = []
+        message_consume_count = 0
+        running = True
+
+        try:
+            while running:
+                message = consumer.poll(timeout=request_timeout)
+
+                if message is not None:
+                    if message.error():
+                        if (
+                            message.error().code()
+                            == confluent_kafka.KafkaError._PARTITION_EOF
+                        ):
+                            # End of partition event, exit consumer
+                            self.logger.warn(
+                                f"{message.topic()} [{message.partition()}] "
+                                f"reached end at offset {message.offset()}"
+                            )
+                            running = False
+                        elif message.error():
+                            raise confluent_kafka.KafkaException(message.error())
+                    else:
+                        messages.append(message.value())
+                        message_consume_count += 1
+
+                        if message_consume_limit:
+                            if message_consume_count >= message_consume_limit:
+                                break
+                else:
+                    self.logger.info(
+                        f"No messages found for topic {topic}; closing consumer..."
+                    )
+                    break
+
+        finally:
+            consumer.close()
+
+        return messages
+
+
+class KafkaBatchProduce(Task):
+    """
+    Task for producing a batch of messages to a Kafka topic.
+
+    Args:
```

Comment:
args like `topic` are required either at init or runtime, could we clarify this in the docstring?